frontend/public/module/k8s/types.ts

```diff
@@ -807,6 +807,8 @@ export type Release = {
   channels?: string[];
 };
 
+export type VersionUpdate = { version?: string; image?: string };
```

Comment:
Hmm.  It occurs to me these aren't actually optional.

Reply:
> Hmm. It occurs to me these aren't actually optional.

Really! I got the type info from here - https://github.com/openshift/api/blob/a1523024209f3122be9e9d53515da470c6a2458b/config/v1/types_cluster_version.go#L700.

It seems the API docs doesn't not have the update type inform here - https://docs.openshift.com/container-platform/4.17/rest_api/config_apis/clusterversion-config-openshift-io-v1.html#status-conditionalupdates-release

Reply:
But for console purposes, both are required when submitting an update patch, no?

Reply:
Never mind.  I read the comments in the first doc.

Reply:
Man, those comments are a lot to process this early in the morning!

Per those comments, if `image` exists, [only it should be passed (no version)](https://github.com/openshift/api/blob/a1523024209f3122be9e9d53515da470c6a2458b/config/v1/types_cluster_version.go#L722C5-L722C87).

For console purposes, I think that means our patch should only include `image`.  @wking  or @LalatenduMohanty, is that right?

Reply:
> Man, those comments are a lot to process this early in the morning!
> 
> Per those comments, if `image` exists, [only it should be passed (no version)](https://github.com/openshift/api/blob/a1523024209f3122be9e9d53515da470c6a2458b/config/v1/types_cluster_version.go#L722C5-L722C87).
> 
> For console purposes, I think that means our patch should only include `image`. @wking or @LalatenduMohanty, is that right?

Right, it's a lot to take in this morning.:). It would be more helpful if the docs specified the [availableUpdates](https://docs.openshift.com/container-platform/4.17/rest_api/config_apis/clusterversion-config-openshift-io-v1.html#status) Type information. Let's wait for responses from @wking @LalatenduMohanty

Reply:
If you have both `image` and `version`, I'd personally set both.  Especially if you showed the `version` to your user, and that's what they selected, including it in your patch will allow the cluster-version operator to ensure that's the version they're heading towards (in case some kind of attack or corruption had tricked the update service into associating a recent version number with an old image pullspec, etc.).